evalml/tests/component_tests/test_oversamplers.py

```diff
@@ -0,0 +1,241 @@
+    if 'NC' in sampler.name:
+        oversampler = sampler(categorical_features=[1], sampling_ratio_dict=sampling_ratio_dict)
+
+    new_X, new_y = oversampler.fit_transform(X, y)
+
+    if sampling_ratio_dict == {}:
+        new_length = 1200
+        num_samples = [800, 200, 200]
+    else:
+        new_length = sum(sampling_ratio_dict.values())
+        num_samples = [800, 300, 300]
+    # check the lengths and sampled values are as we expect
+    assert len(new_X) == new_length
+    assert len(new_y) == new_length
+    value_counts = new_y.to_series().value_counts()
+    assert value_counts.values[1] == value_counts.values[2]
+    np.testing.assert_equal(value_counts.values, np.array(num_samples))
+
+    transform_X, transform_y = oversampler.transform(X, y)
+
+    if data_type == "ww":
+        X = X.to_dataframe().values
+        y = y.to_series().values
+    elif data_type == "pd":
+        X = X.values
+        y = y.values
+
+    np.testing.assert_equal(X, transform_X.to_dataframe().values)
+    np.testing.assert_equal(y, transform_y.to_series().values)
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler, SMOTENCSampler, SMOTENSampler])
+def test_oversample_seed_same_outputs(sampler, X_y_binary):
+    X, y = X_y_binary
+    X = pd.DataFrame(X)
+    y = pd.Series([0] * 90 + [1] * 10)
+
+    samplers = []
+    for seed in [0, 0, 1]:
+        oversampler = sampler(sampling_ratio=1, random_seed=seed)
+        if 'NC' in sampler.name:
+            oversampler = sampler(categorical_features=[1], sampling_ratio=1, random_seed=seed)
+        samplers.append(oversampler)
+
+    # iterate through different indices in samplers
+    # in group 1, first two oversamplers in samplers should be equal
+    # in group 2, calling same oversamplers twice should be equal
+    # in group 3, last two oversamplers in samplers should be different
+    for s1, s2 in [[0, 1], [1, 1], [1, 2]]:
+        X1, y1 = samplers[s1].fit_transform(X, y)
+        X2, y2 = samplers[s2].fit_transform(X, y)
+        if s2 == 2 and sampler != SMOTENSampler:
+            # group 3, SMOTENSampler performance doesn't change with different random states
+            with pytest.raises(AssertionError):
+                pd.testing.assert_frame_equal(X1.to_dataframe(), X2.to_dataframe())
+        else:
+            pd.testing.assert_frame_equal(X1.to_dataframe(), X2.to_dataframe())
+        pd.testing.assert_series_equal(y1.to_series(), y2.to_series())
+
+
+@pytest.mark.parametrize("sampler", [SMOTESampler, SMOTENCSampler, SMOTENSampler])
+def test_sampler_dic_overrides_ratio(sampler):
+    X = np.array([[i for i in range(1000)],
+                  [i % 7 for i in range(1000)],
+                  [0.3 * (i % 3) for i in range(1000)]]).T
+    y = np.array([0] * 800 + [1] * 100 + [2] * 100)
+
+    goal_value_dic = {0: 800, 1: 800, 2: 600}
+    oversampler = sampler(sampling_ratio=1, sampling_ratio_dict=goal_value_dic)
```

Comment:
Do we want some kind of feedback here to let the user of the sampler know that this behavior is occuring?

Reply:
@chukarsten I defaulted to updating the docstring to be clearer about this, but I don't think we need to do any logging or raise warnings for this?